### PR TITLE
Save container

### PR DESCRIPTION
**Pull request by sophiegraham**
- added AddMarkdownFile container

- added AddButton component

- App can now create new files in redux store and keep track of existing files.

### CODE REVIEW

**Comment by sophiegraham** [comment]
Added pre-test script, travis should be working. Per Alex

**Comment by acrankin** [comment]
Your component looks OK but the state management isn't working. You're not saving the file with the current text, each file that gets added is just empty strings.

File: src/containers/markdown/AddMarkdownFile.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import AddButton from '../../components/markdown/AddButton';
+import { addMarkdownFile } from '../../actions/markdownActions';
+
+export default class AddMarkdown extends PureComponent {
+  state = {
+    markdownFiles: [],
+    unsubscribe: null
+  };
+
+  updateState = () => {
```
Comment (mehtaphysical) [inline src/containers/markdown/AddMarkdownFile.js:13]: Seems like this code is out of scope. This is code that would be used to display files

**Comment by mehtaphysical** [comment]
@acrankin i think thats what they want to do. It is creating a new blank file

**Comment by mehtaphysical** [comment]
A few general comments. You have added a bit more to this PR than you need to.

Also, you've changed a bunch of naming stuff. That could have been done in a separate PR.

**Review by acrankin — APPROVED** [review]
